File: euler_1.py
# Euler method: example d^2y/dx^2 = -y
# initial condition 1: y = 1, dydx = 0 ==> cosine
# initial condition 2: y = 0, dydx = 1 ==> sine

import matplotlib.pyplot as plt
x = 0
y = 1
dx = 0.001
dydx = 1
p = []
xx = []
for i in range(300):
    dydx += -y*dx
    y += dydx
    x += dx
    #print(x, y)
    p.append(y)
    xx.append(x)
plt.plot(xx,p)
plt.show()

File: euler_1.cpp
#include <iostream>

using namespace std;

int main(void){
    double x, y, dydx, dx;
    int i;
    x = 0;
    y = 1;
    dx = 0.001;
    dydx = 0;

    for(i=0;i<300;i++){
        dydx += -y*dx;
        y += dydx;
        x += dx;
        cout << x << " " << y << endl;
    }

    return 0;
    
}
